src/test/ui/async-await/nested-async-calls.stderr

```diff
@@ -0,0 +1,42 @@
+error[E0277]: future cannot be sent between threads safely
```

Comment:
Hmm, this is a *lot* of information. I guess I'm going to cc @estebank to try to get some more thoughts about what our ideal output looks like here. I worry that this will overload users.

Reply:
I guess @estebank wrote that they love the direction it's going, earlier, but that was before we rebased atop #65345.

Reply:
I also think that this is quite a lot of information - I think there is some value in tracing the path that the compiler has taken to users, I could imagine a scenario where the `Send` bound was introduced quite far away from where the non-`Send` type was captured. Perhaps for intermediate steps, we could do something like: 

```
note: future is not `Send` as this value is used across an await
  --> $DIR/nested-async-calls.rs:17:5
LL | async fn third() {
   |          ----- in function `third`
LL |     let _a: Outer;
   |         -- has type `third::{{closure}}#0::Outer`
LL |     dummy().await;
   |     ^^^^^^^^^^^^^ await occurs here, with `_a` maybe used later
LL | }
   | - `_a` is later dropped here
note: `third` was invoked by `second`
note: `second` was invoked by `first` which was passed to `require_send`
```

Reply:
> but that was before we rebased atop #65345.

@nikomatsakis: The multi-function error messages were present before the rebase.